### PR TITLE
Feature/netcms 58 query pool

### PR DESCRIPTION
**Pull request by sschmittBt**
Added query pool controller to execute queries from a preloaded pool when calling specific pages

### CODE REVIEW

**Review by remi128 — CHANGES_REQUESTED** [review]
This merge request wants to merge into master, but develop is required

File: src/main/java/de/braintags/netrelay/controller/persistence/DisplayAction.java [hunk 1]
```diff
@@ -58,10 +58,22 @@ protected void handleSubobjectEntityDefinition(RoutingContext context, String en
     handler.handle(Future.failedFuture(new UnsupportedOperationException()));
   }
 
-  protected void handleQuery(IQuery<?> query, String entityName, RoutingContext context, CaptureMap map, IMapper mapper,
-      Handler<AsyncResult<Void>> handler) {
+  protected void handleQuery(IQuery<?> query, String entityName, RoutingContext context, CaptureMap map,
+      IMapper<?> mapper, Handler<AsyncResult<Void>> handler) {
+    int limit;
```
Comment (remi128) [inline src/main/java/de/braintags/netrelay/controller/persistence/DisplayAction.java]: initial assignment from  query.getDataStore().getDefaultQueryLimit() to limit saves one command and sonar won't complain

File: src/main/java/de/braintags/netrelay/controller/persistence/DisplayAction.java [hunk 2]
```diff
@@ -58,10 +58,22 @@ protected void handleSubobjectEntityDefinition(RoutingContext context, String en
...
+    if (map.containsKey(PersistenceController.SELECTION_SIZE_CAPTURE_KEY)) {
+      limit = Integer.parseInt(map.get(PersistenceController.SELECTION_SIZE_CAPTURE_KEY));
+    } else
+      limit = query.getDataStore().getDefaultQueryLimit();
+
+    int offset;
+    if (map.containsKey(PersistenceController.SELECTION_START_CAPTURE_KEY)) {
+      offset = Integer.parseInt(map.get(PersistenceController.SELECTION_START_CAPTURE_KEY));
+    } else
+      offset = 0;
```
Comment (remi128) [inline src/main/java/de/braintags/netrelay/controller/persistence/DisplayAction.java]: see above direct assignment

File: src/main/java/de/braintags/netrelay/controller/querypool/package-info.java [hunk 3]
```diff
@@ -0,0 +1,121 @@
+/**
```
Comment (remi128) [inline src/main/java/de/braintags/netrelay/controller/querypool/package-info.java:1]: please link it into de.braintags.netrelay.controller.package-info

**Review by remi128 — CHANGES_REQUESTED** [review]
tests are running, some minor comments

File: src/main/java/de/braintags/netrelay/controller/persistence/DisplayAction.java (hunk 1 above)
Comment (sschmittBt) [reply src/main/java/de/braintags/netrelay/controller/persistence/DisplayAction.java]: Done

File: src/main/java/de/braintags/netrelay/controller/persistence/DisplayAction.java (hunk 2 above)
Comment (sschmittBt) [reply src/main/java/de/braintags/netrelay/controller/persistence/DisplayAction.java]: Done

File: src/main/java/de/braintags/netrelay/controller/querypool/package-info.java (hunk 3 above)
Comment (sschmittBt) [reply src/main/java/de/braintags/netrelay/controller/querypool/package-info.java:1]: Done

**Pull request merged**